### PR TITLE
Fix `qcfilter.add_test` index

### PR DESCRIPTION
**Pull request by maxwelllevin**
This PR fixes an issue where calling `ds.qcfilter.add_test(...)` with `index=np.array([False])` or `index=np.array([True])` would set the bit to the opposite of what it should be.

- [X] Tests added
- [X] PEP8 Standards or use of linter
- [X] Xarray Dataset or DataArray variable naming follows 'ds' or 'da' naming

### CODE REVIEW

**Review by AdamTheisen — APPROVED** [review]
Thanks @maxwelllevin, this looks good.  I'll wait for @mgrover1 and @zssherman to add a second review and then we can merge in.

**Review by zssherman — APPROVED** [review]
Looks good to me as well! Merging

**Pull request merged**
